### PR TITLE
xtest: pkcs11: Add tests for AES CMAC mechanisms

### PR DESCRIPTION
**Pull request by vchong**
Add tests for CKM_AES_CMAC* mechanisms.

Signed-off-by: Victor Chong <victor.chong@linaro.org>

### CODE REVIEW

File: host/xtest/pkcs11_1000.c [hunk 1]
```diff
@@ -1580,6 +1588,29 @@ static void xtest_pkcs11_test_1007(ADBG_Case_t *c)
...
+	{								\
+		{ CKA_SIGN,	&(CK_BBOOL){CK_TRUE},			\
+				sizeof(CK_BBOOL) },			\
+		{ CKA_VERIFY,	&(CK_BBOOL){CK_TRUE},			\
+				sizeof(CK_BBOOL) },			\
+		{ CKA_CLASS,	&(CK_OBJECT_CLASS){CKO_SECRET_KEY},	\
+				sizeof(CK_OBJECT_CLASS) },		\
+		{ CKA_KEY_TYPE,	&(CK_KEY_TYPE){CKK_AES},		\
+				sizeof(CK_KEY_TYPE) },			\
+		{ CKA_VALUE,	(void *)_key_array,			\
```
Comment (etienne-lms) [inline host/xtest/pkcs11_1000.c:1601]: `(_key_array)`

File: host/xtest/pkcs11_1000.c [hunk 2]
```diff
@@ -1639,7 +1670,25 @@ struct mac_test {
...
+#define CKTEST_CMAC_TEST(key, mecha, input_incr, input, output, incr) {	\
+		.attr_key = key,		\
+		.attr_count = ARRAY_SIZE(key),	\
+		.mechanism = mecha,		\
+		.in_incr = input_incr,		\
+		.in = input,				\
+		.in_len = 0,				\
+		.out = output,				\
+		.out_len = ARRAY_SIZE(output),		\
+		.multiple_incr = incr			\
```
Comment (etienne-lms) [inline host/xtest/pkcs11_1000.c:1682]: add a coma `,`

File: host/xtest/pkcs11_1000.c [hunk 3]
```diff
@@ -1580,6 +1588,29 @@ static void xtest_pkcs11_test_1007(ADBG_Case_t *c)
...
+		{ CKA_SIGN,	&(CK_BBOOL){CK_TRUE},			\
+				sizeof(CK_BBOOL) },			\
+		{ CKA_VERIFY,	&(CK_BBOOL){CK_TRUE},			\
+				sizeof(CK_BBOOL) },			\
+		{ CKA_CLASS,	&(CK_OBJECT_CLASS){CKO_SECRET_KEY},	\
+				sizeof(CK_OBJECT_CLASS) },		\
+		{ CKA_KEY_TYPE,	&(CK_KEY_TYPE){CKK_AES},		\
+				sizeof(CK_KEY_TYPE) },			\
+		{ CKA_VALUE,	(void *)_key_array,			\
+				sizeof(_key_array) }			\
```
Comment (etienne-lms) [inline host/xtest/pkcs11_1000.c:1602]: nitpick: prefer with a coma `,` at end of line.

File: host/xtest/pkcs11_1000.c (hunk 1 above)
Comment (vchong) [reply host/xtest/pkcs11_1000.c:1601]: ok

File: host/xtest/pkcs11_1000.c (hunk 3 above)
Comment (vchong) [reply host/xtest/pkcs11_1000.c:1602]: Ok. Was trying to be consistent with `CK_MAC_KEY_HMAC`. Will add a comma there as well.

File: host/xtest/pkcs11_1000.c (hunk 2 above)
Comment (vchong) [reply host/xtest/pkcs11_1000.c:1682]: Ok. Was trying to be consistent with `CKTEST_MAC_TEST`. Will add a comma there as well.

**Comment by vchong** [comment]
IBART error seems to be a dependency issue on https://github.com/OP-TEE/optee_client/pull/286 and https://github.com/OP-TEE/optee_os/pull/4883.

**Comment by vchong** [comment]
Comments addressed

**Comment by etienne-lms** [comment]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

**Comment by vchong** [comment]
Rebased and tag added.

**Pull request merged**